protocols/imap/src/main/java/org/apache/james/imap/api/message/FetchData.java

```diff
@@ -142,6 +142,16 @@ public boolean getVanished() {
         return vanished;
     }
 
+    public boolean isOnlyFlags() {
+        return bodyElements.isEmpty()
+            && itemToFetch.stream()
+            .filter(item -> item != Item.FLAGS)
```

Comment:
one filter

Reply:
I prefer reading 2 filters, but there might be runtime conciderations so I'll apply your suggestions.

Reply:
What a constructive discussion.